src/Commands/Setup.php

```diff
@@ -146,7 +152,7 @@ protected function writeFile(string $file, string $content): void
             mkdir($directory, 0777, true);
         }
 
-        if (file_exists($path)) {
+        if (file_exists($path) && $file !== 'Config/Security.php') {
```

Comment:
I know this code predates this PR, but I'm just noticing it... `file_exists()` should really only be used to check for collisions. `is_file()` has better performance and won't return `true` for directories (which can cause a lot of confusion, given how the first method is named).

Reply:
I agree with this.
But I think it is more logical for us to solve this issue in a separate PR under title **refactor**.
I think file `Setup.php` needs **refactor**. This refactor can include fixing problem `APPPATH\Config/Auth.php`, fixing the not very interesting naming of methods(`writeFile `and `write_file `| `replace `and `$this->replacer->replace`) and fixing the use of `file_exists()`.
However, if you think that this should be fixed in public relations. please inform.

Reply:
I like your approach; hold for later.